extensions/amp-ooyala-player/0.1/amp-ooyala-player.js

```diff
@@ -0,0 +1,210 @@
+    const pCode = user().assert(
+      this.element.getAttribute('data-pcode'),
+      'The data-pcode attribute is required for <amp-ooyala-player> %s',
+      this.element);
+    const playerId = user().assert(
+      this.element.getAttribute('data-playerid'),
+      'The data-playerid attribute is required for <amp-ooyala-player> %s',
+      this.element);
+    const autoplay = this.element.getAttribute('autoplay');
+
+    let src = 'https://player.ooyala.com/iframe.html?platform=html5-priority';
+    const playerVersion = this.element.getAttribute('data-playerversion') || '';
+    if (playerVersion.toLowerCase() == 'v4') {
+      src = 'https://player.ooyala.com/static/v4/sandbox/amp_iframe/' +
+        'skin-plugin/amp_iframe.html?pcode=' + encodeURIComponent(pCode);
+      const configUrl = this.element.getAttribute('data-config');
+      if (configUrl) {
+        src += '&options[skin.config]=' + encodeURIComponent(configUrl);
+      }
+    }
+
+    src += '&ec=' + encodeURIComponent(embedCode) +
+      '&pbid=' + encodeURIComponent(playerId);
+    if (autoplay) {
+      src += '&autoplay=true';
+    }
+    this.iframe_.src = src;
+
+    window.addEventListener('message',
+                            event => this.handleOoyalaMessages_(event));
+
+    return this.loadPromise(this.iframe_)
+      .then(() => {
+        this.element.dispatchCustomEvent(VideoEvents.LOAD);
+        this.playerReadyResolver_(this.iframe_);
+      });
+  }
+
+  /** @override */
+  unlayoutCallback() {
+    if (this.iframe_) {
+      removeElement(this.iframe_);
+      this.iframe_ = null;
+    }
+    return true;
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  viewportCallback(visible) {
+    this.element.dispatchCustomEvent(VideoEvents.VISIBILITY, {visible});
+  }
+
+  /** @override */
+  pauseCallback() {
+    // Only send pauseVideo command if the player is playing. Otherwise
+    // The player breaks if the user haven't played the video yet specially
+    // on mobile.
+    if (this.iframe_) {
+      this.pause();
+    }
+  }
+
+  /** @private */
+  handleOoyalaMessages_(event) {
+    const data = isObject(event.data) ? event.data : tryParseJson(event.data);
+    if (data === undefined) {
+      return; // We only process valid JSON.
+    }
+    if (data.data == 'playing') {
+      this.element.dispatchCustomEvent(VideoEvents.PLAY);
+    } else if (data.data == 'paused') {
+      this.element.dispatchCustomEvent(VideoEvents.PAUSE);
+    } else if (data.data == 'muted') {
+      this.element.dispatchCustomEvent('mute');
+    } else if (data.data == 'unmuted') {
+      this.element.dispatchCustomEvent('unmute');
+    }
+  }
+
+  // VideoInterface Implementation. See ../src/video-interface.VideoInterface
+
+  /** @override */
+  play(unusedIsAutoplay) {
+    this.playerReadyPromise_.then(() => {
+      this.iframe_.contentWindow./*OK*/postMessage(VideoEvents.PLAY, '*');
```

Comment:
Replace `VideoEvents.PLAY` with string `play` VideoEvents.X are only meant for events and should not be used for methods calls.